Texstyle/Texstyle/ControlStateText.swift

```diff
@@ -0,0 +1,207 @@
+//
+// Copyright (c) 2019 Rosberry. All rights reserved.
+//
+
+import UIKit
+
+/// Stores string, styles and substyles for attributed string depending on ControlState.
+public final class ControlStateText {
+    public let value: String
+    public let styles: [ControlState: TextStyle]
+
+    /// Returns attributed string for normal state.
+    public var attributed: NSAttributedString? {
+       attributed()
+    }
+
+    /// Returns array of subsyles associated with specific controlstate
+    var substyles: [ControlState: [TextSubstyle]] = [:] {
+       didSet {
+           observeStyles()
+       }
+    }
+
+    private var cachedAttributedStrings: [ControlState: NSAttributedString] = [:]
+
+    /// Initialize the text with passed string and style for normal state.
+    ///
+    /// - Parameters:
+    ///   - value: The string for style and substyles.
+    ///   - substyle: The style for passed string.
+    public init(value: String, style: TextStyle) {
+        self.value = value
+        self.styles = [.normal: style]
+        observeStyles()
+    }
+
+    /// Initialize the text with passed string and style for normal state. Returns nil if there is no value.
+    ///
+    /// - Parameters:
+    ///   - value: The string for style and substyles.
+    ///   - substyle: The style for passed string.
+    public convenience init?(value: String?, style: TextStyle) {
+        if let value = value {
+           self.init(value: value, style: style)
+           observeStyles()
+        }
+        else {
+           return nil
+        }
+    }
+
+    /// Initialize the text with passed string and styles for appropriate states.
+    ///
+    /// - Parameters:
+    ///   - value: The string for style and substyles.
+    ///   - styles: The dictionary with states and text styles.
+    public init(value: String, styles: [ControlState: TextStyle]) {
+        self.value = value
+        self.styles = styles
+        observeStyles()
+    }
+
+    /// Initialize the text with passed string and styles for appropriate states. Returns nil if there is no value.
+    ///
+    /// - Parameters:
+    ///   - value: The string for style and substyles.
+    ///   - styles: The dictionary with states and text styles.
+    public convenience init?(value: String?, styles: [ControlState: TextStyle]) {
+        if let value = value {
+           self.init(value: value, styles: styles)
+           observeStyles()
```

Comment:
`observeStyles()` was already called in `init(value: value, styles: styles)`. Should you call it here?